dask/array/linalg.py

```diff
@@ -660,11 +665,17 @@ def compression_matrix(data, q, n_power_iter=0, seed=None):
     mat_h = data.dot(omega)
     for j in range(n_power_iter):
         mat_h = data.dot(data.T.dot(mat_h))
+        if recompute:
+            mat_h = mat_h.persist()
+            wait(mat_h)
     q, _ = tsqr(mat_h)
+    if recompute:
```

Comment:
Just from reading the docstring, I can't decide whether or not I would `recompute=True` would imply a `wait` here. Can you give the motivation for it?

Reply:
We need to remove `data` from the graph of `q` before `data` gets used again in `svd_compressed`.  However, I see now that this is only necessary in the context of `svd_compressed`, so I've moved it there.